src/db/queries.ts

```diff
@@ -1,12 +1,50 @@
-import { db } from "./index.js"
-import { accountTableInAccount } from "./schema.js"
+import { eq } from "drizzle-orm";
+import { db } from "./index.js";
+import { accountTableInAccount } from "./schema.js";
 
 export const createAccount = async (email: string, password: string, mii_name: string, pid: number, pnid: string) => {
-    await db.insert(accountTableInAccount).values({
+    const [account] = await db.insert(accountTableInAccount).values({
         email: email,
         password: password,
         miiName: mii_name,
         pid: pid,
-        pnid: pnid
+        pnid: pnid,
+    }).returning({
+        accountId: accountTableInAccount.accountId,
+        creationDatetime: accountTableInAccount.creationDatetime,
+        email: accountTableInAccount.email,
+        password: accountTableInAccount.password,
+        environment: accountTableInAccount.environment,
+        permissionLevel: accountTableInAccount.permissionLevel,
+        miiName: accountTableInAccount.miiName,
+        pid: accountTableInAccount.pid,
+        pnid: accountTableInAccount.pnid,
+        lastLoginDatetime: accountTableInAccount.lastLoginDatetime,
     });
-}
\ No newline at end of file
+
+    return account;
+};
+
+export const getAccountByEmail = async (email: string) => {
+    const account = await db.select().from(accountTableInAccount).where(eq(accountTableInAccount.email, email));
+
+    return account[0] || null;
```

Comment:
AFAIK, I don't think there's a "find one" function. There is `findMany` but not `findOne`.

Reply:
im pretty sure there is but thats not what im doing here, im getting the first item in the array